### PR TITLE
Fix the Mail Editor not scrolling to the cursor when entering out of view

### LINKED ISSUE #3893: Typing enter doesn't scroll the mail editor.
### To reproduce
1. Create a new email
2. Keep typing enter so that the cursor goes off the bottom of the window
3. See that the editor doesn't scroll to keep the cursor in view

(Typing non-newline characters will cause the view to scroll)

### What should happen?
The cursor should always stay in view

#### Bonus points
When the view scrolls to show the cursor, (after typing other characters), the current line will be positioned at the _very_ bottom of the screen, which is unpleasant to look at imo. It would be nice to make sure that there is always some margin between the cursor and the bottom of the dialog

### PR DESCRIPTION
**Pull request by rezbyte**
Closes #3893.

### CODE REVIEW

File: src/gui/editor/Editor.ts [hunk 1]
```diff
@@ -158,12 +158,43 @@ export class Editor implements ImageHandler, Component {
...
+		})
+
 		this.domElement = domElement
 		// the _editor might have been disabled before the dom element was there
 		this.setEnabled(this.enabled)
 		this.initialized.resolve()
 	}
 
+	// Finds the nearest parent element of the element `element` and returns it
+	private static findParentByClass(element: HTMLElement, targetClass: string) {
```
Comment (charlag) [inline src/gui/editor/Editor.ts:184]: return type is missing

Comment (charlag) [inline src/gui/editor/Editor.ts:184]: may I suggest [Element.closest()](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) ? Maybe you have already considered it, then it probably makes sense to document it

Comment (rezbyte) [reply src/gui/editor/Editor.ts:184]: Thank you, it's name totally slipped my mind. I will use `Element.closest()` instead.

Comment (charlag) [reply src/gui/editor/Editor.ts]: I didn't know it exists so I've also learned something :smile:

**Review by wrdhub — APPROVED** [review]

**Pull request merged**